### DISCUSSION TITLE
James Pearce - TinyBase

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#201] [Announcements]
- https://tinybase.org/
- https://tripleodeon.com/
- https://github.com/jamesgpearce
- https://tinybase.org/guides/releases/#v4-1

### Topics 

- Local First
- [OSS Culture](https://www.youtube.com/watch?v=vBbDAk5WYEo)
- Tech leadership
- Portal

### Questions

- [x] You worked at facebook on open source at a very pivotal time. What projects did you work on? How did that time shape your views on OSS?
- [x] You also helped build Portal. What part of it did you work on?

#### TinyBase

- [x] What do you feel is so novel about local first apps?
- [x] What is tinybase?
- [x] What browser technologies is tinybased built on? Do they have limitations?
- [x] How does tinybase use wasm?
- [x] How do I consume the tinybase data in my react code?
- [x] You recently release [ui-components](https://tinybase.org/demos/ui-components/). Why did you build them and why would I want to use them in my app? Are they headless?
- [x] What developer tools does tinybase provide?
- [x] How does type safety work?
- [x] Does tiny base support relation data?
- [x] TinyQL is neat. What are some unique aspects to it?
- [x] How does tiny base integrate with crdts?

#### Future

- [ ] There a growing divide in apps local vs cloud. Do you think theres merits to could app? Or should we try to build towards a more local first future?

**Comment by hipstersmoothie** [comment]
https://github.com/biomejs/biome
https://jazz.tools/

**Reply by jamesgpearce** [reply]
Three areas of tips:

Interesting dev dependencies:
[typedoc](https://typedoc.org) - massive amount of documentation in the dts files, can parse and extract examples to run in jest
[asciichart](https://github.com/kroitor/asciichart), performance graphs
[ts-unused-exports](https://www.npmjs.com/package/ts-unused-exports)


Personal productivity:
Keynote!
 - continuous video across slides
 - supports SVG
VS Code:
 - [SVG plugin](https://marketplace.visualstudio.com/items?itemName=jock.svg), with preview & minify
 - [Geodata viewer](https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.geo-data-viewer), GPX and route editing


On the boat:
Lots of cool stuff on a boat, best left for another podcast
12v power system, so anything that powers off a cigaret lighter is awesome; anything that requires the step-up inverter is bad:
 - DeWalt power tools all charged off 12
 - great fast charging usb adaptors so power all laptops direct from 12
 - latest little a monitor, ViewSonic VA1655. 16 Inch 1080p, runs off USBC, so can use it as second display, or a small TV for gaming or watching stuff off phone. A TV without mains power is very cool :) - and only $125 or something

**Comment by just-be-dev** [comment]
- https://www.scattered-thoughts.net/writing/babys-first-wasm-compiler
- https://multi.app/
